addons/sale_stock/static/src/widgets/qty_at_date_widget.js

```diff
@@ -40,9 +40,9 @@ export class QtyAtDateWidget extends Component {
         this.popover = usePopover(this.constructor.components.Popover, { position: "top" });
         this.orm = useService("orm");
         this.calcData = {};
-        onWillRender(async () => {
+        onWillStart(async () => {
             await this.initCalcData();
-        })
+        });
```

Comment:
```suggestion
        });
        onWillUpdateProps(async () => {
            await this.initCalcData();
        });
```
I think this would be a good idea to also add it on the `onWillUpdateProps` hook (as well as the `onWillStart`), so it will properly trigger its changes when exiting the edition of the line rather than waiting for the save of the whole record to get re-rendered.